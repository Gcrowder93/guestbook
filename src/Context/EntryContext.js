import { useContext, createContext, useState } from 'react';

const EntryContext = createContext();

export function EntryProvider({ children }) {
  const [guestEntry, setguestEntry] = useState([]);
  const [date, setDate] = useState([]);
  const contextValue = { guestEntry, setguestEntry, date, setDate };
  return <EntryContext.Provider value={contextValue}>{children}</EntryContext.Provider>;
}

export function useEntry() {
  const context = useContext(EntryContext);
  if (context === undefined) {
    throw new Error('error undefined error');
  }
  return context;
}
