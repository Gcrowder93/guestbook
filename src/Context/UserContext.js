import { createContext, useContext, useState } from 'react';

const UserContext = createContext();

export function UserProvider({ children }) {
  const [userInput, setUserInput] = useState('');

  // const value = useMemo(() => ({ userInput, setUserInput }), [userInput]);
  const contextValue = { userInput, setUserInput };

  return <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>;
}

export function useUser() {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('error undefined error');
  }

  return context;
}
