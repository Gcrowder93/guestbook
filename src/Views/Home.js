import EntryForm from '../Components/EntryForm/EntryForm.js';
import EntryList from '../Components/EntryList/EntryList.js';

const Home = () => {
  return (
    <>
      <EntryForm />
      <EntryList />
    </>
  );
};

export default Home;
