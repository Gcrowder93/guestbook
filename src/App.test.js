import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
// import App from './App';
import { EntryProvider } from './Context/EntryContext';
import { UserProvider } from './Context/UserContext';
import Home from './Views/Home';

test('renders learn react link', () => {
  render(
    <EntryProvider>
      <UserProvider>
        <Home />
      </UserProvider>
    </EntryProvider>
  );
  const linkElement = screen.getByText(/add an entry/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders the actions', async () => {
  render(
    <EntryProvider>
      <UserProvider>
        <Home />
      </UserProvider>
    </EntryProvider>
  );

  const button = screen.getByRole('button', { name: /submit/i });
  const entryText = screen.getByPlaceholderText(/message/i);
  const userInput = screen.getByPlaceholderText(/name/i);
  userEvent.type(entryText, 'hello');
  userEvent.type(userInput, 'chase');
  userEvent.click(button);
  const entry = await screen.findByText(/user/i);
  // expect(entry).toBeInTheDocument();
  expect(entry).toMatchSnapshot();
});
