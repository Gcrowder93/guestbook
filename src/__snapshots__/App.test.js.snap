// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the actions 1`] = `
<h1>
  user: chase
    
</h1>
`;
